### PR TITLE
Documenting most of OpenStack Taster in YARD

### PR DESCRIPTION
**Pull request by Codysseus**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+`openstack_taster` was designed to make it easier for future developers to add features easily.
+
+1. Flags  
+If you want to add a flag to the tool, add an `opts.on` statement to the `OptionParser` block.
+Here you can specify the short and long version of the flag as well as the description.
+A block can be added to `opts.on` so that once it parses the flag, any code inside the block will run.
+
+2. Test Suites  
+If you want to add a test suite to `openstack_taster`, then there are only 3 things you need to do:
+  1. Add a suite name and description to the `suites` hash. Do not use spaces.
```
Comment (knightsamar) [inline CONTRIBUTING.md]: You will need 4 spaces to make the sub-level formatting work.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  2. Add this line in `lib/openstack_taster.rb:taste`: `return_values.push taste_<suite_name>(<parameters>) if settings[:<suite_name>]`  
+      `<suite_name>` is the name of your suite defined in the `suites` hash in `bin/openstack_taster`.
+  3. Add your function to `lib/openstack_taste.rb`. The standard for suite function names is `taste_<suite_name>`.
```
Comment (knightsamar) [inline CONTRIBUTING.md]: `openstack_taster.rb` - *r* is missing.

File: lib/openstack_taster.rb [hunk 3]
```diff
@@ -191,12 +216,19 @@ def error_log(logger, level, message, dup_stdout = false, context = nil)
...
+  # @param instance [Fog::Compute::OpenStack::Server] the instance to query
+  # @return [String] the name of the image
   def get_image_name(instance)
     @image_service
       .get_image_by_id(instance.image['id'])
       .body['name']
   end
 
+  # Create an image of an instance.
+  # @note This method blocks until the image is complete on the server.
```
Comment (knightsamar) [inline lib/openstack_taster.rb]: do you mean "snapshot creation" is complete on the server ?

File: lib/openstack_taster.rb [hunk 4]
```diff
@@ -52,8 +53,21 @@ def initialize(
...
   end
 
-  def taste(image_name, settings) # FIXME: Reduce Percieved and Cyclomatic complexity
-    image = @compute_service.images # FIXME: Images over compute service is deprecated
+  # Taste a specified image
+  # @param image_name [String] The name on OpenStack of the image to be tested.
+  # @param settings [Hash] A hash of settings to enable and disable tests, snapshot creation upon failure.
+  # @return [Boolean] success or failure of tests on image.
+  # @note The testing section could be further streamlined by:  
+  #   creating a naming standard for test functionions (i.e. taste_<name>)
```
Comment (knightsamar) [inline lib/openstack_taster.rb]: "functions" :)

File: lib/openstack_taster.rb [hunk 5]
```diff
@@ -128,6 +142,11 @@ def taste(image_name, settings) # FIXME: Reduce Percieved and Cyclomatic complex
     end
   end
 
+  # Runs the security test suite using inspec
+  # @param instance [Fog::Image::OpenStack::Image] The instance to test.
+  # @param username [String] The username to use when logging into the instance.
+  # @return [Boolean] Whether or not the image passed hte security tests.
+  # @todo Don't crash when connection refused.
```
Comment (knightsamar) [inline lib/openstack_taster.rb:149]: Does it really crash ? I never saw this.

File: lib/openstack_taster.rb [hunk 6]
```diff
@@ -52,8 +53,21 @@ def initialize(
...
+  # @param image_name [String] The name on OpenStack of the image to be tested.
+  # @param settings [Hash] A hash of settings to enable and disable tests, snapshot creation upon failure.
+  # @return [Boolean] success or failure of tests on image.
+  # @note The testing section could be further streamlined by:  
+  #   creating a naming standard for test functionions (i.e. taste_<name>)  
+  #   limiting the parameters of each test to be: instance, distro_username  
+  #   Adding a 'suites' subhash to the settings hash  
+  #   Then that subhash can be iterated over, use eval to call each function,
+  #   appending the suite name to 'taste_' for the function name
+  #   and passing the standardized parameters.
```
Comment (knightsamar) [inline lib/openstack_taster.rb:66]: These ideas are already done, right ?  We probably just need to document it as a standard in the CONTRIBUTING doc.

**Review by knightsamar — CHANGES_REQUESTED** [review]

File: lib/openstack_taster.rb (hunk 6 above)
Comment (Codysseus) [reply lib/openstack_taster.rb:66]: The 'taste_<suite>' is already done and documented. This is just a suggestion to make it even easier to add a suite to the tester.

File: lib/openstack_taster.rb (hunk 5 above)
Comment (Codysseus) [reply lib/openstack_taster.rb:149]: That was a comment left by @tolvstaa. I never saw this either.

File: lib/openstack_taster.rb (hunk 3 above)
Comment (Codysseus) [reply lib/openstack_taster.rb]: This was another @tolvstaa coment, but I will rephrase it that way.

**Comment by Codysseus** [comment]
@knightsamar Fixed!

**Review by knightsamar — APPROVED** [review]
Let's tackle the `tries` constant in a seperate PR and merge this for now.

**Pull request merged**